### PR TITLE
Push promise requests won't have have authorization header

### PR DESCRIPTION
**Pull request by jitsni**
Push promise requests won't have have authorization header (instead
authorization bit mask is sent in the DATA frame)

### CODE REVIEW

**Comment by dpwspoon** [comment]
I think we need a test (or 2) where when a push promise has the authorization header removed, the response to that push promise MUST have the private cache-control header set (if the cache-control header on the response did not have public set)

**Comment by dpwspoon** [comment]
I think we need a test that shows we aren't propagating responses that introduce a vary header to push promises that have now removed their vary header.  Likewise, if there is a vary header that matches, we probably need to forward that header on the request.

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/edge-arch/polling.vary.header.mismatch/connect/client.rpt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+connected
+
+write close
+
+read nukleus:begin.ext ${http:headers()
+                             .item(":status", "200")
+                             .item("content-length", "11")
+                             .item("date", response1Date)
+                             .item("surrogate-control", "max-age=1+100")
+                             .item("etag", etag)
```
Comment (dpwspoon) [inline src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/edge-arch/polling.vary.header.mismatch/connect/client.rpt:47]: I think this is missing `.item("vary", "user-agent")`, there are other things missing on a few of the vary tests, but I think the other tests (cache-control private for authentication) looks good.

For the vary header, we still need to solve for push promises that are late in the wrap around from h2, and have had their headers stripped.  Potentially to null values that match the stored response but not the wrapped around response.  Additionally, we need to have a test for Vary: "*"

[Link to spec](https://tools.ietf.org/html/rfc7231#section-7.1.4): https://tools.ietf.org/html/rfc7231#section-7.1.4

**Review by dpwspoon — CHANGES_REQUESTED** [review]

**Review by jfallows — APPROVED** [review]

File: src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/edge-arch/polling.vary.header.mismatch/connect/client.rpt (hunk 1 above)
Comment (jitsni) [reply src/main/scripts/org/reaktivity/specification/nukleus/http_cache/streams/proxy/edge-arch/polling.vary.header.mismatch/connect/client.rpt:47]: Fixed vary header tests and also added a test for vary: *

**Review by jfallows — APPROVED** [review]

**Comment by jfallows** [comment]
All feedback has been applied.

**Pull request merged**